src/main/java/de/leonhard/storage/internal/settings/DataType.java

```diff
@@ -2,40 +2,45 @@
 
 import de.leonhard.storage.internal.provider.LightningProviders;
 import de.leonhard.storage.internal.provider.MapProvider;
-import java.util.Map;
 import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
 
 /**
  * An Enum defining how the Data should be stored
  */
 @RequiredArgsConstructor
 public enum DataType {
-  SORTED {
-    @Override
-    public Map<String, Object> getMapImplementation() {
-      return mapProvider.getSortedMapImplementation();
-    }
-  },
 
-  UNSORTED {
-    @Override
-    public Map<String, Object> getMapImplementation() {
-      return mapProvider.getSortedMapImplementation();
-    }
-  };
+    // Sorted data type
+    SORTED {
+        @Override
+        public @NotNull Map<String, Object> getMapImplementation() {
+            return mapProvider.getSortedMapImplementation();
+        }
+    },
+
+    // Unsorted data type
+    UNSORTED {
+        @Override
+        public @NotNull Map<String, Object> getMapImplementation() {
+            return mapProvider.getSortedMapImplementation();
```

Comment:
Why is the unsorted implementation also created by calling `mapProvider.getSortedMapImplementation()`. You could also just do `new HashMap<>()` if I am not mistaken.